src/StoreKeeper/WooCommerce/B2C/Exports/OrderExport.php

```diff
@@ -670,35 +679,110 @@ protected function catchKnownExceptions($throwable)
+        $paymentModule = $this->storekeeper_api->getModule('PaymentModule');
         // Attach refunds to order
         if (PaymentGateway::hasUnsyncedRefunds($woocommerceOrderId)) {
             $refundPayments = PaymentGateway::getUnsyncedRefundsPaymentIds($woocommerceOrderId);
 
             foreach ($refundPayments as $refundPayment) {
-                $refundPaymentId = $refundPayment['payment_id'];
-                $refundId = $refundPayment['refund_id'];
+                $refundPaymentId = $refundPayment['sk_refund_id'];
+                $refundId = $refundPayment['wc_refund_id'];
+                $refundAmount = $refundPayment['amount'];
                 $this->debug(
                     'Attaching payment refund to order',
-                    ['order_id' => $storekeeper_id, 'payment_id' => $refundPaymentId]
+                    ['order_id' => $storekeeperId, 'payment_id' => $refundPaymentId]
                 );
 
                 try {
-                    $ShopModule->attachPaymentIdsToOrder(['payment_ids' => [$refundPaymentId]], $storekeeper_id);
+                    $shopModule->attachPaymentIdsToOrder(['payment_ids' => [$refundPaymentId]], $storekeeperId);
+                    PaymentGateway::markRefundAsSynced($woocommerceOrderId, $refundPaymentId, $refundId);
                 } catch (\Throwable $generalException) {
                     $this->debug('Refund was not attached to order', [
-                        'order_id' => $storekeeper_id,
+                        'order_id' => $storekeeperId,
                         'payment_id' => $refundPaymentId,
                     ]);
+                    PaymentGateway::scheduleRefundTask($woocommerceOrderId, $refundId, $refundAmount);
                 }
-                PaymentGateway::markRefundAsSynced($WpObject->get_id(), $refundPaymentId, $refundId);
             }
         }
 
+        $unsyncedRefundsWithoutPaymentIds = PaymentGateway::getUnsyncedRefundsWithoutPaymentIds($woocommerceOrderId);
+        if (count($unsyncedRefundsWithoutPaymentIds) > 0) {
+            foreach ($unsyncedRefundsWithoutPaymentIds as $unsyncedRefundsWithoutPaymentId) {
+                $payOrderRefundId = false;
+                try {
+                    $id = $unsyncedRefundsWithoutPaymentId['id'];
+                    $refundId = $unsyncedRefundsWithoutPaymentId['wc_refund_id'];
+                    $refundAmount = $unsyncedRefundsWithoutPaymentId['amount'];
+                    if (PaymentGateway::hasPayment($woocommerceOrderId)) {
+                        $storekeeperPaymentId = PaymentGateway::getPaymentId($woocommerceOrderId);
+                        $storekeeperRefundId = $shopModule->refundAllOrderItems([
+                            'id' => $storekeeperId,
+                            'refund_payments' => [
+                                [
+                                    'payment_id' => $storekeeperPaymentId,
+                                    'amount' => -abs($refundAmount),
```

Comment:
```suggestion
                                    'amount' => round(-abs($refundAmount),2),
```
Otherwise you might miss one cent

Reply:
Added `round`.